[EVAL] Italian Big Math Expression

danielepoterti opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, **failure to follow the guidelines below will result in the PR being closed automatically**. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access be granted. 🚨

**PLEASE READ THIS**:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject it since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. **Starting April 10, the minimum eval count is 15 samples, we hope this makes it easier to create and contribute evals.**

Also, please note that we're using **Git LFS** for storing the JSON files, so please make sure that you move the JSON file to Git LFS before submitting a PR. Details on how to use Git LFS are available [here](https://git-lfs.com).

## Eval details 📑

### Eval name

Italian big math expression

### Eval description

This test aims to assess the model's ability to solve written mathematical expressions in Italian involving large numbers and provide the answers in written form.

### What makes this a useful eval?

Testing the model's linguistic and numerical capabilities in Italian involves not only an understanding of the language but also the ability to accurately solve complex mathematical expressions. This cross-domain functionality - spanning both linguistic and numerical domains - is a stringent test of the model's abilities.

Moreover, providing the answer in a written format further pushes the model, assessing its ability to convert numerical values into written language. This is an essential feature for numerous applications, such as voice assistants which need to express numerical information audibly, or transcription software that transcribes numbers into words for the visually impaired.

In previous tests, it's been observed that GPT 3.5 struggles to accurately translate large Italian numbers into their correct written form. For instance, the number 10,029,202 was translated as 'diecimiladuecentonovantaduemiladuecentodue', which is incorrect in Italian.

Another fascinating evaluation could be to test the model's ability to translate individual digits into written Italian words.

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can create an eval on cases where the model fails to reason about the physical world.
- [x] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [x] **Include at least 15 high-quality examples.**

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

> Insert what makes your eval high quality that was not mentioned above. (Not required)

## Eval structure 🏗️

Your eval should

- [x] Check that your data is in `evals/registry/data/{name}`
- [x] Check that your YAML is registered at `evals/registry/evals/{name}.yaml`
- [x] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (<https://platform.openai.com/docs/usage-policies>).

- [x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the commits on the merged pull request.

- [x] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgment

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and the high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access be granted.

### Submit eval

- [x] I have filled out all required fields of this form
- [x] I have used **Git LFS** for the Eval JSON data
- [ ] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "cinquecentoventitré per centosessantacinque più trecentosessantaquattro per duecentotrentanove"}], "ideal": "centosettantatrémiladuecentonovantuno"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "novecentoventotto per quattrocentodiciannove meno cinquecentoquattordici per trecentonovantadue"}], "ideal": "centottantasettemilatrecentoquarantaquattro"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "seicentosettantotto per quattrocentosettantuno più cinquecentoventi per duecentonovanta"}], "ideal": "quattrocentosettantamilacentotrentotto"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "novecentoquindici per trecentocinquantacinque meno seicentoventisei per duecentosettanta"}], "ideal": "centocinquantacinquemilaottocentocinque"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "ottocentoventiquattro per cinquecentotrentasette più cinquecentonovanta per duecentoventisette"}], "ideal": "cinquecentosettantaseimilaquattrocentodiciotto"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "settecentocinquantacinque per quattrocentoquarantotto meno cinquecentoquattordici per trecentoquaranta"}], "ideal": "centosessantatrémilaquattrocentottanta"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "novecentosettantatré per seicentosettantacinque più cinquecentodiciassette per duecentosettantacinque"}], "ideal": "settecentonovantottomilanovecentocinquanta"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "ottocentosettantasei per trecentoquarantacinque meno seicentoventisei per duecentoventinove"}], "ideal": "centocinquantottomilaottocentosessantasei"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "novecentocinquantanove per quattrocentocinquantasette più cinquecentonovanta per duecentoventisette"}], "ideal": "cinquecentoquarantanovemilaquattrocentonovantatré"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "settecentosessantasette per cinquecentoventi meno cinquecentoquattordici per trecentoquarantasei"}], "ideal": "duecentoventimilanovecentonovantasei"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "ottocentocinquantanove per seicentocinquantanove più cinquecentodiciotto per duecentosettantatré"}], "ideal": "settecentosettemilaquattrocentonovantacinque"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "novecentonovantatré per trecentocinquantotto meno seicentoventisei per duecentoventuno"}], "ideal": "duecentodiciassettemilacentoquarantotto"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "ottocentotrentasette per cinquecentocinquantanove più cinquecentodiciotto per duecentosessantacinque"}], "ideal": "seicentocinquemilacentocinquantatré"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "novecentotrentaquattro per quattrocentoventinove meno cinquecentoquattordici per trecentoquarantasei"}], "ideal": "duecentoventiduemilaottocentoquarantadue"}
{"input": [{"role": "system", "content": "La risposta dovrebbe includere solo il risultato dell'operazione scritto in forma di parole, in minuscolo, senza alcun carattere aggiuntivo o spiegazioni aggiuntive."}, {"role": "user", "content": "settecentosettantotto per cinquecentonovantauno più cinquecentoventi per duecentoventi"}], "ideal": "cinquecentosettantaquattromilacentonovantotto"}
  ```
</details>

usama-openai reviewed (changes requested):
Thanks for the contribution. I would like to request some changes.

I would recommend asking the model to reason before answering and using the `Includes` evaluation method here because arithmetic operations and other complex tasks are hard for the model to do zero-shot without a chance to reason through the steps. Asking the model to provide reasoning will give the model a fair chance to solve the question. You can add instructions to provide the chain of thought before answering and to provide the final output in a specific format, and then use that format to write the ideal answer. For example, the model can be asked to enclose the final answer in square brackets, and the ideal answer can be formatted like `[cinquecentoundicimilacentosettantatré]`. It'll give the model a proper chance to reason before answering, and the proper formatting will help in identifying the final answer using the `Includes` method.

We would love to review the PR again after the suggested changes.

danielepoterti commented:
@usama-openai thank you for your feedback on my previous submission. I've taken your suggestions into consideration and implemented the necessary changes.

Specifically, I have enhanced the model to incorporate a reasoning step prior to providing answers. Additionally, I've modified the prompt to require a specific answer format and shifted the evaluation technique to the 'Includes' method.

I would appreciate it if you could review the updated PR at your earliest convenience. I remain open to further feedback and I'm ready to make additional adjustments if necessary.

Kindly let me know if there are any further changes you would like me to make.

usama-openai reviewed (approved):
This PR looks in good shape now. I'm approving this PR.
